lightning/src/util/ser.rs

```diff
@@ -383,6 +383,35 @@ impl Readable for BigSize {
 	}
 }
 
+/// The lightning protocol uses u16s for lengths in most cases. As our serialization framework
+/// primarily targets that, we must as well. However, because we may serialize objects that have
+/// more than 65K entries, we need to be able to store larger values. Thus, we define a variable
+/// length integer here which is backwards-compatible but treats 0xffff as "read eight more bytes".
+struct U16Or64(pub u64);
+impl Writeable for U16Or64 {
```

Comment:
This kinda looks like `U160r64`. Would prefer to name this either `CollectionLength` or at least `U16OrU64` which is a little easier to read.

Reply:
Strongly agree with Jeff. I only ever read it as 160r64 and didn't realize there was an `Or` in there until Jeff's comment. I wasn't sure what either `160` or `r` meant, but didn't even think to ask. The `O` of `Or` remains rather unfortunate.

Reply:
lol I had to Google it before realizing it was not a zero. Not sure it's worth putting the implementation details into the name, tbh.